lib/sensors/INA_226.h

```diff
@@ -0,0 +1,57 @@
+#ifndef INA_226_H
+#include INA_226_H
+#include "mbed.h"
+#include "math.h"
+#include "rover_config.h"
+
+// Datasheet for INA226 http://www.ti.com/lit/ds/symlink/ina226.pdf
+// current sensor data https://docs.google.com/spreadsheets/d/1Qn4QbEJ1Ia54vYnTgaTvLxHralQvCz6PEGaSF_O1L20/edit#gid=0
+
+//configuration register 00h controls modes of chip
+// bit 15 - reset bit, wriitng 1 resets registers. bits 0-2 select operating mode
+// bus voltage register 02h
+// power register 03h
+//calibration register 05h
+// mask/enable register 06h
+//alert limit register 07h
+
+class INA_226{
+    private:  
+        constexpr u_int8_t CONFIG_REGISTER 0x00;
+        constexpr u_int8_t VOLTAGE_REGISTER 0x01;
+        constexpr u_int8_t CURRENT_REGISTER 0x04;
+        constexpr u_int8_t CALIBRATION_REGISTER 0x05;
+
+        //  I2C bus initialization
+        I2C i2c_;
+        //may need to adjust function to take sensor slave addresses as there many be many current sensors
+        u_int8_t sensor_address_;
+
+        enum CONFIGURATIONMODE{
+            powerReset = 0,
+            triggered = 3,
+            continous = 7
+        };
+
+        //below are settings for the INA226 chip. The default values have been selected 
+        //but different conversion times can be selected as well. See datasheet for other settings
+        constexpr int BUS_VOLTAGE_CONVERSION_TIME = 4; //sets the conversion time for bus voltage measurement(1.1 ms)
+        constexpr int SHUNT_VOLTAGE_CONVERSION_TIME = 4; //sets the conversion time for shunt voltage measurement(1.1ms)
+        constexpr int AVERAGE_MODE = 0;
+
+        //current conversion constexprants
+        float max_expected_current_ = 0; 
+        float current_lsb = max_expected_current_ / pow(2,15); 

```

Comment:
this will not work as expected as you will set max_expected_current_  in the constructor. Also usually you don't initialize the class members that will be set by the constructor. If you want default values, then use default argument or constructor overloading.

Reply:
What should I do for the Constants I have defined? Do I have them as a normal variable and then set them in them in the constructor?

Reply:
yes